### ISSUE TITLE
Multiple viewer instances on the same HTML page

### ISSUE THREAD

**Issue by pdoorr** [stephanrauh/ngx-extended-pdf-viewer#654]
I need to show many pdf on the same page then I've added ngx-extended-pdf-viewer for each one. The problem is that only the first viewer "works" but it loads the pdf of the latest one, the remaining viewers are empty and give me errors... Is there a way to make it work?

Thanks,
fabio

**Comment by stephanrauh** [comment]
I'm afraid it's not possible. I almost never say no... but in this case, it's really next to impossible.

The problem is that pdf.js uses global variables, and I never bothered to change that. Not that I'd got a chance. pdf.js is the PDF viewer of Firefox, which by definition never displays more than one PDF at the same time.

However, what you can do - probably, I haven't tried it myself yet - is to put ngx-extended-pdf-viewer in an iFrame. That allows to to display multiple instances side-by-side, each wrapped into the own iFrame.

Basically, that's also the approach of ng2-pdfjs-viewer. Until recently, I'd have happily sent you to that library, but I'm not sure it's still alive. So I reluctantly suggest wrapping ngx-extended-pdf-viewer in an iFrame, and I'd love to hear from you. Does it work? Which problems did you encounter? Can I help you achieving your goal?

**Comment by pdoorr** [comment]
Hi Stephan,
many thanks for your prompt answer. I've changed my approach enabling only
one viewer per time.

Wonderful job with the viewer!

Best regards,
fabio

Il giorno mar 16 feb 2021 alle ore 23:24 Stephan Rauh <
notifications@github.com> ha scritto:

**Comment by stephanrauh** [comment]
Hi Fabio,

glad to hear you've found a solution! It's just a pity that this means I still don't know how to display multiple PDF files side by side... it seems I have to research that myself. :)

Let's close the issue for the moment. It's a popular feature, so I won't forget it.

Best regards,
Stephan

**Comment by achrysanthakopoulou** [comment]
Hello there! I understand that this is a really difficult issue, but do you have any plans on working on this?
This is the only reason I'm not using your library..
I tried this: 
```
<iframe *ngIf="source" height="100%" width="100%">
   <ngx-extended-pdf-viewer [textLayer]="true" [showHandToolButton]="true" [src]="source"></ngx-extended-pdf-viewer>
</iframe>
```
but got the error "offsetParent not set - cannot scroll"

P.S. I don't want to use ng2-pdfjs-viewer because I don't see many collaborators/updates. My alternative is a plain iframe but this does not work on tablet/mobile browsers.
Thanks!

**Comment by stephanrauh** [comment]
It seems I've forgotten to update all the tickets I've closed before finding a solution... look here:

https://pdfviewer.net/extended-pdf-viewer/side-by-side

https://github.com/stephanrauh/extended-pdf-viewer-showcase/tree/main/src/app/extended-pdf-viewer/side-by-side 

As for ng2-pdfjs-viewer: yeah, that's a pity. Several years I've managed to contact the owner of the library, but he's stopped answering, so I don't have a clue about the state of the project.

**Comment by achrysanthakopoulou** [comment]
Thank you for your answer! So the issue of having multiple instances of ngx-extended-pdf-viewer is officially resolved by using iframes?
If so, I cannot understand how to accomplish it.
Is your github example:
```
<iframe id="embedded" src="/extended-pdf-viewer/iframe/cdk" style="height: 100vh; width: 47%"></iframe>
<iframe id="embedded" src="/extended-pdf-viewer/iframe/graalvm" style="height: 100vh; width: 47%; float: right"></iframe>
```
There are only two iframe without any reference to ngx-extended-pdf-viewer.
1. I cannot understand what the source is.
2. SideBySideComponent does not seem to do anything related to the iframe.

Thanks!

**Comment by achrysanthakopoulou** [comment]
Oh thanks! It seems a bit hacky so maybe I'll stick with the classic iframe for the time being!
